dataprofiler/tests/profilers/test_column_profile_compilers.py

```diff
@@ -70,6 +71,132 @@ def test_add_profilers(self):
+
+        
+        # Test same data types
+        data1 = pd.Series(['-2', '15', '1', '2'])
+        data2 = pd.Series(['5', '-1'])
+
+        compiler1 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data1)
+        compiler2 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data2)
+        expected_diff = {
+            'data_type_representation': {
+                'datetime': 'unchanged',
+                'int': 'unchanged',
+                'float': 'unchanged',
+                'string': 'unchanged'
+            },
+             'data_type': 'unchanged',
+             'statistics': {
+                     'min': -1.0,
+                     'max': 10.0,
+                     'sum': 12.0,
+                     'mean': 2.0,
+                     'variance': 38.666666666666664,
+                     'stddev': 3.285085839971525
+                 }
+        }
+
+        self.assertDictEqual(expected_diff, compiler1.diff(compiler2))
+        
+        # Test different compilers
+        data1 = pd.Series(['-2', '-1', '1', '2'])
+        data2 = pd.Series(['5', '15'])
+
+        compiler1 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data1)
+        compiler2 = col_pro_compilers.ColumnStatsProfileCompiler(data2)
+        # Assert type error is properly called
+        with self.assertRaises(TypeError) as exc:
+            compiler1.diff(compiler2)
+        self.assertEqual(str(exc.exception),
+                         "`ColumnPrimitiveTypeProfileCompiler` and "
+                         "`ColumnStatsProfileCompiler` are not of the same "
+                         "profile compiler type.")
+        
+    def test_disabling_columns_during_primitive_diff(self):
+        data1 = pd.Series(['-2', '-1', '1', '2'])
+        data2 = pd.Series(['5', '15'])
+        options = StructuredOptions()
+
+        # Test disabled column in one compiler
+        options.int.is_enabled = False
+        compiler1 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data1,
+                                                                         options)
+        compiler2 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data2)
+        expected_diff = {
+            'data_type_representation': {
+                'datetime': 'unchanged', 
+                'float': 'unchanged', 
+                'string': 'unchanged', 
+                'int': [None, 1.0]}, 
+            'data_type': ['float', 'int']
+        }
+        self.assertDictEqual(expected_diff, compiler1.diff(compiler2))
+
+        # Test disabled column in both compilers
+        compiler2 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data2,
+                                                                         options)
+        expected_diff = {
+            'data_type_representation': {
+                'datetime': 'unchanged', 
+                'float': 'unchanged', 
+                'string': 'unchanged'
```

Comment:
as above, `int` should be [None, None]

Reply:
If both values are none than nothing has changed exactly, no need to include it

Reply:
imo `unchanged` is different from `[None, None]`. Unchanged is when we have type but the type is unchanged, but [None, None] means none of profiles has the int option enabled. [None, None] should be consistent with the format that I mentioned above.

Reply:
If the options suggest to leave it out of the profiler in both cases, I think it is fair to not have it in the diff.

Example, user turns off everything but the labeling in two profiles. We shouldn't give them information on int / float in the diff

Reply:
At best, I could see `None` to maintain consistency. but not a list